### ISSUE TITLE
Video freezes in VODs. Audio unaffected.

### ISSUE THREAD

**Issue by BlueSialia** [crackededed/Xtra#325]
I was using the version 2.14.x and updated to the latest 2.21.0 hoping it would fix it, but no luck.

In both versions the video freezes while watching a VOD after less than a minute. Audio keeps playing as usual.

I think it happens faster if the quality is higher.

**Comment by crackededed** [comment]
i'm guessing the device just doesn't support those qualities. you could try playing a vod in another video player app and see if it works fine. here's a link to a vod:
https://dgeft87wbj63p.cloudfront.net/805b8d805688069c18c4_pgl_dota2_40320976392_1673508923/chunked/index-dvr.m3u8

**Comment by SamuelBucher** [comment]
> i'm guessing the device just doesn't support those qualities. you could try playing a vod in another video player app and see if it works fine. here's a link to a vod: https://dgeft87wbj63p.cloudfront.net/805b8d805688069c18c4_pgl_dota2_40320976392_1673508923/chunked/index-dvr.m3u8

Yeah, this one has the same issue. Interestingly, the regular Twitch app plays videos just fine. I suppose this issue is on my (or our) end.

**Comment by crackededed** [comment]
i'm not sure if there's anything that can be done for devices that lag or have issues with exoplayer even when using a hardware decoder. maybe some other player library would work but i don't know if it could be made to work with any of the existing code

**Comment by Phylaprae** [comment]
Hello, 
I don't know why but the issue I mentioned here https://github.com/crackededed/Xtra/issues/545 happen less now.
I think there is something with video + chat pane. When I remove the chat, framerate can improve.

**Comment by nightznero** [comment]
> Hello, I don't know why but the issue I mentioned here [#545](https://github.com/crackededed/Xtra/issues/545) happen less now. I think there is something with video + chat pane. When I remove the chat, framerate can improve.

i also agree,on some updates vod is tragedy to watch,and on some is rare to being annoyed.
Its not about exoplayer then,its must be something diff

**Comment by crackededed** [comment]
there's a new test build you can try where you can change the player to mpv in debug settings:

https://github.com/crackededed/Xtra/actions?query=branch%3Ampv

https://github.com/crackededed/Xtra/actions/runs/20559446660/artifacts/4979938004
